applications/GeoMechanicsApplication/custom_workflows/dgeosettlement.h

```diff
@@ -77,13 +77,10 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) KratosGeoSettlement
     {
         if (!GetComputationalModelPart().HasNodalSolutionStepVariable(rVariable)) return;
 
-        VariableUtils{}.SetHistoricalVariableToZero(rVariable, GetComputationalModelPart().Nodes());
-
-        block_for_each(GetComputationalModelPart().Nodes(),
-                       [&rVariable, SourceIndex, DestinationIndex](auto& node) {
-            node.GetSolutionStepValue(rVariable, DestinationIndex) =
-                node.GetSolutionStepValue(rVariable, SourceIndex);
-        });
+        NodeUtilities::AssignUpdatedVectorVariableToNonFixedComponentsOfNodes(
+            GetComputationalModelPart().Nodes(), rVariable, ZeroVector{3}, 0);
+        NodeUtilities::AssignUpdatedVectorVariableToNonFixedComponentsOfNodes(
+            GetComputationalModelPart().Nodes(), rVariable, ZeroVector{3}, 1);
```

Comment:
And then the same thing here:
```suggestion
            GetComputationalModelPart().Nodes(), rVariable, rVariable.Zero(), 1);
```

Reply:
Done